File: backend/src/main/java/dev/frilly/locket/controller/IndexController.java
package dev.frilly.locket.controller;

import dev.frilly.locket.Versioning;
import dev.frilly.locket.dto.res.VersionResponse;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * The controller for handling on route ("/").
 */
@RestController
public final class IndexController {

  /**
   * GET /: Retrieves the backend's version.
   */
  @GetMapping("/")
  public VersionResponse getIndex() {
    return new VersionResponse(Versioning.getCurrent().toString());
  }

}

File: backend/src/main/kotlin/dev/frilly/locket/controller/IndexController.kt
package dev.frilly.locket.controller

import dev.frilly.locket.LocketBackendApplication
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

/**
 * The controller for handling on route ("/").
 */
@RestController
class IndexController {

    data class VersionResponse(val version: String)

    /**
     * Retrieves the API version of the backend.
     */
    @GetMapping("/")
    fun getIndex(): VersionResponse {
        return VersionResponse(LocketBackendApplication.API_VERSION)
    }

}